src/main/java/org/folio/dew/batch/bulkedit/jobs/processidentifiers/BulkEditInstanceIdentifiersJobConfig.java

```diff
@@ -39,14 +46,16 @@
 public class BulkEditInstanceIdentifiersJobConfig {
   private final BulkEditInstanceProcessor bulkEditInstanceProcessor;
   private final BulkEditSkipListener bulkEditSkipListener;
+  private final BulkEditMarcProcessor bulkEditMarcProcessor;
 
   @Bean
   public Job bulkEditProcessInstanceIdentifiersJob(JobCompletionNotificationListener listener, Step bulkEditInstanceStep,
-    JobRepository jobRepository) {
+    Step bulkEditMarcStep, JobRepository jobRepository) {
     return new JobBuilder(ExportType.BULK_EDIT_IDENTIFIERS + JOB_NAME_POSTFIX_SEPARATOR + INSTANCE.getValue(), jobRepository)
       .incrementer(new RunIdIncrementer())
       .listener(listener)
       .flow(bulkEditInstanceStep)
+      .next(bulkEditMarcStep)
```

Comment:
Hi @obozhko-folio , seems here instances are retrieved twice. The same with file with identifier reading. If it is true, is it possible to do this once using spring batch logic?

Reply:
Yes, it reads the file twice, but to configure reading only once, it needs to store all identifiers in the memory after the first read. I don't think it's a good idea.